circuits/cpp/barretenberg/py/src/estimation/protogalaxy.py

```diff
@@ -0,0 +1,48 @@
+from circuit import CircuitKZG
+from flavors import GoblinUltra
+from goblin import Goblin
+from msm import MSM
+from utils import *
+
+
+class FoldingVerifier(CircuitKZG):
+    # k instances folded into 1
+    def __init__(self, circuits):
+        flavor = GoblinUltra()
+        self.log_n_in_stack = circuits[0].log_n
+        num_instances_to_fold = len(circuits)  # k
+        num_msms = flavor.NUM_POLYNOMIALS
+        # don't need to fold shifts?
+        num_msms -= flavor.NUM_SHIFTED_POLYNOMIALS
+        # our constants include NUM_WITNESSES-many id polynomials
+        num_msms -= flavor.NUM_WITNESSES
+        # our constants include 2 Lagrange polynoma
+        num_msms -= 2
+        self.msms = [MSM(num_instances_to_fold) for _ in range(num_msms)]
+        # rough estimate for number of non-MSM gates
+        constant_overhead = 500 + 5 * self.log_n_in_stack
```

Comment:
Would be very helpful to put these magic numbers in a constants file that describes where the estimates come from